torch/_dynamo/eval_frame.py

```diff
@@ -691,138 +696,174 @@ def result_capturing_wrapper(*graph_inputs):
-            # `kwargs`. The former can be safely ignored. Because Dynamo.export does not
-            # export them as part of the function signature. The latter will be handled
-            # in the next step.
-            for unprovided_arg in fullargspec.args[
-                len(args) : -len(fullargspec.defaults or [])
-            ]:
-                assert unprovided_arg in kwargs, f"Missing argument {unprovided_arg}"
-
-        # 4. Keyword arguments provided in `kwargs`.
-        input_strs += list(kwargs.keys())
-
-        # 5. Keyword-only arguments with default values if not provided are not exported
-        # as part of the function signature.
-        for kwonly_arg in fullargspec.kwonlyargs:
-            kwonlydefaults = fullargspec.kwonlydefaults or {}
-            assert (
-                kwonly_arg in kwargs or kwonly_arg in kwonlydefaults
-            ), f"Missing keyword only argument {kwonly_arg}"
-
-        return input_strs
-
-    new_graph.graph._codegen = _PyTreeCodeGen(
-        _PyTreeInfo(
-            argument_names(f, *args, **kwargs),
-            in_spec,
-            out_spec_traced,
+            def placeholder(self, target, args, kwargs):
+                arg = next(self.old_args_gen)
+                if "val" in self.current_node.meta:
+                    arg.node.meta["val"] = self.current_node.meta["val"]
+                if "tensor_dict" in self.current_node.meta:
+                    arg.node.meta["tensor_dict"] = self.current_node.meta["tensor_dict"]
+                return arg
+
+            def output(self, target, args, kwargs):
+                dynamo_result_flat = args[0]
+                lookup = [*dynamo_result_flat, *self.new_args]
+                new_result_flat = [lookup[i] for i in matched_output_elements_positions]
+                return super().output(target, (new_result_flat,), {})
+
+            def run_node(self, n):
+                self.current_node = n
+                r = super().run_node(n)
+                if "val" in self.current_node.meta:
+                    r.node.meta["val"] = self.current_node.meta["val"]
+                return r
+
+        if aten_graph:
+            # Running graph with interpreter is needed for propagating the stack_trace
+            def graph_with_interpreter(*args):
+                with torch.fx.traceback.preserve_node_meta():
+                    return torch.fx.Interpreter(graph).run(*args)
+
+            graph = make_fx(
+                graph_with_interpreter,
+                decomposition_table=decomposition_table,
+                tracing_mode=tracing_mode,
+                _allow_non_fake_inputs=True,
+            )(*graph_captured_input)
+
+        new_graph = ChangeInputOutputSignature(
+            graph,
+        ).transform()
+
+        # Make dynamo graph to have same input/output spec as user code
+        def argument_names(f: Callable[..., Any], *args, **kwargs) -> List[str]:
```

Comment:
NIT: i feel like this try block is too big now. Could we make these kind of things into separate utility methods so that the code is little shorter?

Reply:
Not in this pr